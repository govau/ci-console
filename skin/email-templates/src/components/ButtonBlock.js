import React from "react";
import styled from "styled-components";

import Table, { Td } from "./Table";
import { default as BaseA } from "./A";

const Block = styled(Table)`
  min-width: 100%;
  border-collapse: collapse;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
`;

const BlockInner = styled(Td)`
  padding-top: 0;
  padding-right: 18px;
  padding-bottom: 18px;
  padding-left: 18px;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
`;

const ContentContainer = styled(Table)`
  border-collapse: separate;
  border-radius: 3px;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  background-color: ${({ theme }) => theme.button.primary.color};
`;

const Content = styled(Td)`
  font-size: 16px;
  padding: 15px;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  font-family: ${({ theme }) => theme.button.fontFamily};
`;

const A = styled(BaseA)`
  font-weight: normal;
  letter-spacing: normal;
  line-height: 100%;
  text-align: center;
  text-decoration: none;
  color: #ffffff;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  display: block;
`;

const ButtonBlock = ({ children, ...rest }) => {
  return (
    <Block border="0" cellPadding="0" cellSpacing="0" width="100%">
      <tbody>
        <tr>
          <BlockInner align="center" valign="top">
            <ContentContainer border="0" cellPadding="0" cellSpacing="0">
              <tbody>
                <tr>
                  <Content align="center" valign="middle">
                    <A title={children} target="_blank" {...rest}>
                      {children}
                    </A>
                  </Content>
                </tr>
              </tbody>
            </ContentContainer>
          </BlockInner>
        </tr>
      </tbody>
    </Block>
  );
};

export default ButtonBlock;
