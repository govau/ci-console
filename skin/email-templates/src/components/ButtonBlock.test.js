import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import ButtonBlock from "./ButtonBlock";

test("renders without crashing", () => {
  const tree = renderer.create(
    <ThemeProvider
      theme={{
        button: {
          fontFamily: "Roboto",
          primary: {
            color: "#ffcc00"
          }
        }
      }}
    >
      <ButtonBlock href="https://example.com">content</ButtonBlock>
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});
