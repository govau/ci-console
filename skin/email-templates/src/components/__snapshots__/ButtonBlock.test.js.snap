// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.c1 {
  mso-table-lspace: 0pt;
  mso-table-rspace: 0pt;
}

.c0 {
  min-width: 100%;
  border-collapse: collapse;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
}

.c2 {
  padding-top: 0;
  padding-right: 18px;
  padding-bottom: 18px;
  padding-left: 18px;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
}

.c3 {
  border-collapse: separate;
  border-radius: 3px;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  background-color: #ffcc00;
}

.c4 {
  font-size: 16px;
  padding: 15px;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  font-family: Roboto;
}

.c5 {
  font-weight: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 100%;
  text-align: center;
  text-decoration: none;
  color: #ffffff;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  display: block;
}

<table
  border="0"
  cellPadding="0"
  cellSpacing="0"
  className="c0 c1"
  width="100%"
>
  <tbody>
    <tr>
      <td
        align="center"
        className="c2"
        valign="top"
      >
        <table
          border="0"
          cellPadding="0"
          cellSpacing="0"
          className="c3 c1"
        >
          <tbody>
            <tr>
              <td
                align="center"
                className="c4"
                valign="middle"
              >
                <a
                  className="c5 "
                  href="https://example.com"
                >
                  content
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;
